clients/src/main/java/org/apache/kafka/common/record/LazyDownConversionRecordsSend.java

```diff
@@ -24,35 +24,36 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.channels.GatheringByteChannel;
+import java.util.Iterator;
 
 /**
  * Encapsulation for {@link RecordsSend} for {@link LazyDownConversionRecords}. Records are down-converted in batches and
- * on-demand when {@link #writeRecordsTo} method is called.
+ * on-demand when {@link #writeTo} method is called.
  */
 public final class LazyDownConversionRecordsSend extends RecordsSend<LazyDownConversionRecords> {
     private static final Logger log = LoggerFactory.getLogger(LazyDownConversionRecordsSend.class);
     private static final int MAX_READ_SIZE = 128 * 1024;
 
-    private RecordsProcessingStats processingStats = null;
-    private RecordsWriter convertedRecordsWriter = null;
-    private LazyDownConversionRecords.Iterator convertedRecordsIterator;
+    private RecordConversionStats processingStats = null;
```

Comment:
nit: change name as well. Maybe just `stats` since we're in a conversion context?